scripts/us_epa/parent_company/process_parent_company.py

```diff
@@ -0,0 +1,298 @@
+_OWNERSHIP_CLEAN_CSV_HDR = (_DCID, _EPA_FACILITY_GHG_ID, _YEAR,
+                            _PERCENT_OWNERSHIP)
+
+_COUNTERS = {
+    "missing_zip": set(),
+    "percent_ownership_not_found": set(),
+    "facility_does_not_exist": set(),
+    "company_name_not_found": set(),
+    "year_does_not_exist": set(),
+}
+
+
+def _gen_mcf():
+    lines = [
+        "Node: dcid:EpaOrganizationOwnership",
+        "description: The ownership of an EPA Facility by an Organization in a given year.",
+        "typeOf: dcs:StatisticalVariable", "populationType: dcs:Organization",
+        "measuredProperty: owns", "statType: measurementResult"
+    ]
+    return "\n".join(lines)
+
+
+def _gen_ownership_tmcf():
+    lines = [
+        "Node: E:EpaParentCompanyOwnership->E0",
+        "typeOf: dcs:StatVarObservation",
+        "variableMeasured: dcid:EpaOrganizationOwnership",
+        f"observationDate: C:EpaParentCompanyOwnership->{_YEAR}",
+        f"observationAbout: C:EpaParentCompanyOwnership->{_DCID}",
+        f"value: C:EpaParentCompanyOwnership->{_EPA_FACILITY_GHG_ID}",
+        f"ownershipPercentage: C:EpaParentCompanyOwnership->{_PERCENT_OWNERSHIP}",
+    ]
+    return "\n".join(lines)
+
+
+def _gen_company_tmcf():
+    lines = [
+        "Node: E:EpaParentCompanyTable->E1", "typeOf: dcs:Organization",
+        f"{_DCID}: C:EpaParentCompanyTable->{_DCID}",
+        f"{_NAME}: C:EpaParentCompanyTable->{_NAME}",
+        f"{_ADDRESS}: C:EpaParentCompanyTable->{_ADDRESS}",
+        f"{_CIP}: C:EpaParentCompanyTable->{_CIP}"
+    ]
+    return "\n".join(lines)
+
+
+def _v(table, row, col):
+    return row.get(_TABLE_PREFIX + "." + table + "." + col, "")
+
+
+def _cv(table, row, col):
+    return _v(table, row, col).strip().title()
+
+
+def _get_name(table, row):
+    name = _cv(table, row, "PARENT_COMPANY_NAME")
+    return name.replace(" Llc", " LLC")
+
+
+def _name_to_id(s):
+    s = sub(r'\W+', '', s)
+    s = s.replace(' Llc', ' LLC')
+    return ''.join([s[0].upper(), s[1:]])
+
+
+def _get_address(table, row):
+    parts = []
+    for k in ["PARENT_CO_STREET_ADDRESS", "PARENT_CO_CITY", "PARENT_CO_STATE"]:
+        p = _cv(table, row, k)
+        if p:
+            parts.append(p)
+    address = ", ".join(parts)
+    p = _cv(table, row, "PARENT_CO_ZIP")
+    if p:
+        address += " - " + p
+    return address
+
+
+def _get_county(company_id, zip, year):
+    """Resolve the geo relations for the given Facility
+
+    Returns resolved <zip>, <county>
+    """
+    if zip == "zip/00000" or zip == "zip/":
+        _COUNTERS["missing_zip"].add((company_id, year))
+        return ""
+
+    county_candidates = fh.get_county_candidates(zip)
+    if not county_candidates:
+        _COUNTERS["missing_zip"].add((company_id, year))
+        return ""
+
+    # Choose the first.
+    try:
+        # This could fail because county_candidates is of form [[], []]
+        county = county_candidates[0][0]
+    except:
+        county = ""
```

Comment:
did you inspect these failures to see if these were expected? were there other cases that were missed?

Reply:
Yup! Ended up making a few edits and the code documentation now describes when we cannot find a county.